### DISCUSSION TITLE
Use value of R variable in Callout Box Caption with User Interface/Visual Quarto.

### DISCUSSION THREAD

**Discussion by mlamias** [quarto-dev/quarto-cli#6486] [Q&A]
### Description

I am trying to display a callout box that contains an R Code chunk with a caption that is dynamically set by the value of an R variable in my environment.  I am able to achieve this without any problem using the **source** editor in R Studio with the following code:

````qmd

```{r}
mycaption<-"This is my custom caption"
```

This example places the value of mycaption into the callout box caption:

::: callout-note
## `r mycaption`

```{r}
#| eval: false
#| code: y<-LETTERS
#| echo: true

```
:::

````

This results in the following (and is exactly what I am trying to achieve):

![image](https://github.com/quarto-dev/quarto-cli/assets/9097516/d30c7954-f8d3-4a49-8d0c-25c2d41e83f1)

Now, I am wondering if I can achieve the same result using the **visual** editor or IDE in R studio.  When I switch back to the visual editor and click on the and click the three dots (...) on the .callout-note it displays the following:

![image](https://github.com/quarto-dev/quarto-cli/assets/9097516/a91c2713-6578-4912-9f6f-6d820ede830f)

If I modify the caption text box to be r mycaption2 and then switch back to the visual editor I see the code has been modified as follows:

````qmd

::: callout-note
## r mycaption2

```{r}
#| eval: false
#| code: y<-LETTERS
#| echo: true

```
:::
````

As you can see the backticks have been removed from the callout-not caption (## r mycaption2).  Is there a way to enter a r variable name into the **visual** editor caption so that backticks will appear in the **visual** editor as (## \`r mycaption2\`) so that the value of the value of the r variable will appear in my callout box caption, resulting in the same output as shown in my first output screenshot above?

**Comment by mcanouil** [comment]
I let @jjallaire answer as he is the one with the best knowledge on the visual editor.

**Reply by mlamias** [reply]
Thanks.  Yes, basically, I'm hoping there is some what to simply add \`r mycaption2\` to the visual editor caption text box with the text being escaped in the source editor.

**Reply by mcanouil** [reply]
To note, inline expression in Quarto 1.4 (currently in pre-release state) is bringing a new homogenous inline expression syntax which will "replace" the `knitr` specific syntax (_i.e._, `{r} x`), see <https://quarto.org/docs/prerelease/1.4/inline>.

**Reply by mlamias** [reply]
Thank you.  This is great!

Also, as a side note, I see there's a typo in the link you provided.  Under [Markdown Input](https://quarto.org/docs/prerelease/1.4/inline#markdown-output), it reads:

#Not that by default, the output of inline expressions is treated as ordinary text...

But should read:

#Not**e** that by default, the output of inline expressions is treated as ordinary text...

The "e" seems to be missing on the what should be the word "Note."

**Reply by mcanouil** [reply]
Thanks, I made a PR to fix the typo:

- quarto-dev/quarto-web#806
